SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/PartitionManager.cpp

```diff
@@ -311,6 +363,123 @@ void FPartitionManager::Impl::Flush(ISpatialOSWorker& Connection)
 	}
 }
 
+void FPartitionManager::Impl::FlushInterestUpdates(ISpatialOSWorker& Connection)
+{
+	if (!InterestUpdate.bEnabled)
+	{
+		return;
+	}
+
+	for (auto Worker : ConnectedWorkers)
+	{
+		if (!Worker->State->bInterestInitialized)
+		{
+			ChangeInterestRequest Request;
+			Request.SystemEntityId = Worker->State->SystemWorkerId;
+			Request.bOverwrite = false;
+
+			{
+				// Initial snapshot interest query.
+				ChangeInterestQuery SnapshotQuery;
+				SnapshotQuery.Entities.Add(SpatialConstants::INITIAL_GLOBAL_STATE_MANAGER_ENTITY_ID);
+				SnapshotQuery.Entities.Add(SpatialConstants::INITIAL_SPAWNER_ENTITY_ID);
+				SnapshotQuery.Entities.Add(SpatialConstants::INITIAL_VIRTUAL_WORKER_TRANSLATOR_ENTITY_ID);
+				SnapshotQuery.Components = InterestF.GetServerNonAuthInterestResultType().ComponentIds;
+				SnapshotQuery.ComponentSets = InterestF.GetServerNonAuthInterestResultType().ComponentSetsIds;
+				Request.QueriesToAdd.Add(SnapshotQuery);
+			}
+
+			if (PartitionView.GetCompleteEntities().Num() > 0)
+			{
+				ChangeInterestQuery PartitionQuery;
+				PartitionQuery.Entities.Append(PartitionView.GetCompleteEntities());
+				PartitionQuery.Components = InterestUpdate.PartitionMetaData;
+				PartitionQuery.Components.Add(SpatialConstants::PARTITION_ACK_COMPONENT_ID);
+				Request.QueriesToAdd.Add(MoveTemp(PartitionQuery));
+			}
+			if (SystemWorkersDispatcher.SubView.GetCompleteEntities().Num() > 0)
+			{
+				ChangeInterestQuery WorkerQuery;
+				WorkerQuery.Entities.Append(SystemWorkersDispatcher.SubView.GetCompleteEntities());
+				WorkerQuery.Components = { SpatialConstants::WORKER_COMPONENT_ID, SpatialConstants::SYSTEM_COMPONENT_ID };
+				Request.QueriesToAdd.Add(MoveTemp(WorkerQuery));
+			}
+			if (WorkersDispatcher.SubView.GetCompleteEntities().Num() > 0)
+			{
+				ChangeInterestQuery ServerWorkerQuery;
+				ServerWorkerQuery.Entities.Append(WorkersDispatcher.SubView.GetCompleteEntities());
+				for (auto Entity : ServerWorkerQuery.Entities)
```

Comment:
minor: would prefer non auto type.

Reply:
coding standard says autos should be used for range based iterators

Reply:
This is super pedantic but it only says that for iterators: https://docs.unrealengine.com/4.27/en-US/ProductionPipelines/DevelopmentSetup/CodingStandard/#the'auto'keyword feel free to ignore.

Reply:
Oh that's fair - I've always read it as valid in any range-based iteration